File: _2024/day-01/problem 1/problem1.py
with open("c:/Users/ppmpr/OneDrive/Documents/GitHub/AOC24/day 1/problem1/input.txt", 'r') as file:
    data = file.read()

list_1 = []
list_2 = []

for i in data.split("\n"):
    parts = i.split()
    if len(parts) >= 2:
        list_1.append(parts[0])
        list_2.append(parts[1])

list_1.sort()
list_2.sort()

new_list = []
for i in range(len(list_1)):
    new_list.append(abs(int(list_1[i])-int(list_2[i])))

print(sum(new_list))

File: _2024/day-01/problem 1/problem1.rs
use std::fs;

fn main() {
    let data: String = fs::read_to_string("C:/Users/ppmpr/OneDrive/Documents/GitHub/AOC24/day-01/problem 1/input.txt").expect("Error reading file");
    let mut list1: Vec<i32> = Vec::new();
    let mut list2: Vec<i32> = Vec::new();
    
    for i in data.split("\n") {
        let temp: Vec<&str> = i.trim().split("   ").collect();
        if temp.len() == 2 {
            list1.push(temp[0].parse().unwrap());
            list2.push(temp[1].parse().unwrap());
        }
    }

    list1.sort();
    list2.sort();

    let mut new_lst : Vec<i32> = Vec::new();
    for i in 0..list1.len() {
        new_lst.push(i32::abs(list1[i] - list2[i]));
    }

    println!("{}", new_lst.iter().sum::<i32>());
}